### PR TITLE
Feature: Duration parameter for the load function

### PR DESCRIPTION
**Pull request by mspae**
### Short description of changes:
Adds an optional duration parameter to `wavesurfer.load` which can be used to explicitly set the duration of the audio (for both backends). This is useful to render the correct wave size in cases where the audio is not decoded:

WebAudio backend with peaks provided currently can't know the duration of the audio and thus always renders it at 100% width initially and then jumps to the correct size on first interaction (when audio is decoded) –
 this can be fixed by using the duration parameter (See one use case in #1210)

### Breaking in the external API:
/

### Breaking changes in the internal API:
/

### Todos/Notes:
- Subsequent calls of `load` overwrite the previous duration
- Once the audio is decoded (a buffer is available in `WebAudio.buffer`) its real length is used

### Related Issues and other PRs:
- Improved duration parameter feature initially proposed in #1210 
- Is required to implement lazy-loading of audio element (see #1237 )
- Contains #1233 in order to be able to build

### CODE REVIEW

File: src/wavesurfer.js [hunk 1]
```diff
@@ -1022,10 +1022,12 @@ export default class WaveSurfer extends util.Observer {
      * Loads audio and re-renders the waveform.
      *
      * @param {string} url The url of the audio file
-     * @param {?number[]|number[][]} peaks Wavesurfer does not have to decode the audio to
-     * render the waveform if this is specified
+     * @param {?number[]|number[][]} peaks Wavesurfer does not have to decode
+     * the audio to render the waveform if this is specified
      * @param {?string} preload (Use with backend `MediaElement`)
      * `'none'|'metadata'|'auto'` Preload attribute for the media element
+     * @param {?number} duration The duration of the audio. This is required to
```
Comment (thijstriemstra) [inline src/wavesurfer.js]: I'm not a fan of the word `required` here as this is an optional thing. Could you rephrase this a little?

Comment (mspae) [reply src/wavesurfer.js]: Done. I've tried to explain it a little more in the docs and removed the param docs on the private methods for brevity.

**Comment by mspae** [comment]
@thijstriemstra bumped them to `2.0.0-beta03`

**Comment by thijstriemstra** [comment]
Cool. Branch still has conflicts.
